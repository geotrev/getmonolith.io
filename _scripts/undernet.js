import Undernet from "../lib/esm"

document.addEventListener("DOMContentLoaded", () => Undernet.start())
