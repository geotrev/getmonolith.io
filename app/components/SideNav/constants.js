import { docPages } from "app/routes"

const {
  introduction,
  download,
  css,
  javascript,
  grid,
  typography,
  buttons,
  forms,
  modals,
  collapsibles,
  accordions,
  dropdowns,
  tooltips,
  alignment,
  offsetOrder,
  text,
  display,
  color,
  spacing,
} = docPages

export const NAV_DATA = [
  {
    header: "Overview",
    links: [
      { name: "Introduction", url: introduction },
      { name: "Download", url: download },
      { name: "CSS", url: css },
      { name: "JavaScript", url: javascript },
    ],
  },
  {
    header: "Layout",
    links: [
      { name: "Flex Grid", url: grid },
      { name: "Alignment", url: alignment },
      { name: "Offset / Order", url: offsetOrder },
      { name: "Spacing", url: spacing },
    ],
  },
  {
    header: "Elements",
    links: [
      { name: "Buttons", url: buttons },
      { name: "Forms", url: forms },
      { name: "Typography", url: typography },
    ],
  },
  {
    header: "Components",
    links: [
      { name: "Collapsibles", url: collapsibles },
      { name: "Accordions", url: accordions },
      { name: "Dropdowns", url: dropdowns },
      { name: "Modals", url: modals },
      { name: "Tooltips", url: tooltips },
    ],
  },
  {
    header: "Utilities",
    links: [
      { name: "Color", url: color },
      { name: "Display", url: display },
      { name: "Text", url: text },
    ],
  },
]
