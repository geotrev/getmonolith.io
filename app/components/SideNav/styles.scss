@import "app/global";

// stylelint-disable

#side-nav {
  background: #fbfbfb;

  .side-nav-wrapper {
    padding: 0;

    .side-nav-expand {
      background: $white;

      @include net-breakpoint(xsmall) {
        border-bottom: 1px solid $gray900;
      }

      @include net-breakpoint(xlarge) {
        border-bottom: none;
      }

      > button {
        width: 100%;
        padding: 20px;
        text-decoration: none;
        margin: $global-border-width;
        border: none;
        color: $primary-color;

        &.rotate-180 {
          svg {
            transform: rotate(180deg);
          }
        }

        svg {
          @include net-set-flow("margin-right", 8px);
        }
      }
    }

    .version-text {
      font-size: 14px;
      padding-bottom: 8px;
    }

    .side-nav-menu {
      h4 {
        color: $black;
        font-weight: $global-font-weight-bold;
      }

      ul {
        list-style-type: none;
        padding: 0;
        margin: 0;
        flex: 1 1 auto;

        li {
          padding: 0;
        }

        li a {
          @include colored-link;
          display: block;
          font-size: 16px;
          line-height: 36px;
          text-decoration: none;

          svg {
            @include net-set-flow("margin-left", 4px);
            transition: 0.1s ease-out;
            color: $gray800;
          }

          &:hover,
          &:active,
          &:focus {
            svg {
              @include net-set-flow("margin-left", 8px);
              color: $primary-color;
            }
          }

          &.active {
            color: $primary-color !important;
          }
        }
      }

      &[data-accordion] {
        border: none;
        border-radius: 0;

        [data-target] {
          @include colored-link($black);
          background: transparent;
          padding: 8px 0;
          font-weight: bold;
        }

        .accordion-content {
          background: transparent;
          border: none;

          a {
            @include net-set-flow("padding-left", 8px);
            margin: 2px;
          }
        }
      }
    }
  }
}
