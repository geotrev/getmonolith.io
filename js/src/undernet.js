"use strict"

import Modal from "./components/modal"
import Accordion from "./components/accordion"
import Utils from "./utils"

const Modals = new Modal()
const Accordions = new Accordion()
const Utilities = new Utils()

const Undernet = {
  Modals,
  Accordions,
  Utilities,
}

Undernet.start = () => {
  Undernet.Modals.start()
  Undernet.Accordions.start()
}

Undernet.stop = () => {
  Undernet.Modals.stop()
  Undernet.Accordions.stop()
}

window.Undernet = Undernet
export default Undernet
