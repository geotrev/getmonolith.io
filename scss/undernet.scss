/*!
* @license MIT (https://github.com/geotrev/undernet/blob/master/LICENSE)
* Undernet v2.2.0 (https://undernet.io)
* Copyright 2017-2018 George Treviranus
*/

@import "helpers/functions";
@import "config";
@import "helpers/mixins";

// remove globals if you are scoping!
@import "elements/reset";

// .#{$scope} {
@import "helpers/classes";
@import "elements/grid";
@import "elements/typography";
@import "elements/link";
@import "elements/button";
@import "elements/form";

@import "components/modal";
@import "components/accordion";
// }

// Modals need .no-scroll to be out of $scope in order to function properly.

@if $scope != "" {
  .no-scroll {
    overflow: hidden;
  }
}
