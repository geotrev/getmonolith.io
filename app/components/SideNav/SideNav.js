import React, { useState, useEffect } from "react"
import classNames from "classnames"
import throttle from "lodash/throttle"
import { NavLink } from "react-router-dom"
import Menu from "react-feather/dist/icons/menu"
import ChevronRight from "react-feather/dist/icons/chevron-right"

import { NAV_DATA } from "./constants"
import { Accordions } from "undernet"
import "./styles.scss"

const pkg = require("projectRoot/package.json")
const MENU_COLLAPSE_WIDTH = 1199

export default function SideNav() {
  // set up effects and state

  const getWindowInnerWidth = () => window.innerWidth

  const isLargerThanCollapseWidth = getWindowInnerWidth() > MENU_COLLAPSE_WIDTH
  const [menuIsOpen, setMenuIsOpen] = useState(isLargerThanCollapseWidth)

  let handleMenuVisibility = () => {
    if (isLargerThanCollapseWidth) {
      setMenuIsOpen(true)
    }
  }

  handleMenuVisibility = throttle(handleMenuVisibility, 50)

  const componentUnmountFunction = () => {
    window.removeEventListener("resize", handleMenuVisibility)
    Accordions.stop()
  }

  const observedStateOnMount = []
  useEffect(() => {
    Accordions.start()
    window.addEventListener("resize", handleMenuVisibility)
    setMenuIsOpen(isLargerThanCollapseWidth)

    return componentUnmountFunction
  }, observedStateOnMount)

  const observedStateOnUpdate = [menuIsOpen]
  useEffect(() => {
    Accordions.stop()
    Accordions.start()
  }, observedStateOnUpdate)

  // set up handlers and other helper methods

  const handleCollapseClick = () => {
    if (getWindowInnerWidth() <= MENU_COLLAPSE_WIDTH) {
      setMenuIsOpen(false)
    }
  }

  const handleMenuToggleClick = event => {
    event.preventDefault()
    setMenuIsOpen(!menuIsOpen)
  }

  const buttonClasses = () => {
    return classNames(
      "has-justify-content-center has-align-items-center has-display-flex has-display-none-xlarge",
      {
        "rotate-180": !menuIsOpen,
      }
    )
  }

  const menuClasses = () => {
    return classNames("row side-nav-menu accordion has-padding-3", {
      "has-display-none": !menuIsOpen,
    })
  }

  const accordionIsActive = items => {
    let isActive = false

    items.forEach(item => {
      if (window.location.href.includes(item.url)) {
        isActive = true
      }
    })

    return isActive
  }

  // render content

  const renderAccordionChildLink = item => {
    return (
      <li key={item.name} role="none">
        <NavLink
          role="listitem"
          className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
          activeClassName="active"
          onClick={handleCollapseClick}
          to={item.url}
        >
          {item.name} <ChevronRight size={16} role="presentation" focusable="false" />
        </NavLink>
      </li>
    )
  }

  const renderAccordionRow = (section, index) => {
    const listItems = section.links.map(renderAccordionChildLink)

    return (
      <React.Fragment key={section.header}>
        <h4 className="paragraph">
          <button
            id={`nav-acc-button${index}`}
            data-parent="side-nav-accordion"
            className="accordion-button"
            data-target={`nav-acc-content${index}`}
          >
            {section.header}
          </button>
        </h4>
        <ul className="accordion-content" id={`nav-acc-content${index}`}>
          {listItems}
        </ul>
      </React.Fragment>
    )
  }

  const renderNavAccordion = () => {
    return NAV_DATA.map((section, i) => {
      return (
        <div
          data-visible={accordionIsActive(section.links) ? "true" : "false"}
          data-accordion-row={`nav-acc-content${i}`}
          className="accordion-row is-xsmall-12 column has-no-padding"
          key={section.links[0].url}
        >
          {renderAccordionRow(section, i)}
        </div>
      )
    })
  }

  return (
    <div className="is-xsmall-12 is-xlarge-2 column has-no-padding" id="side-nav">
      <div className="is-fluid grid side-nav-wrapper">
        <div className="row has-display-flex has-display-none-xlarge side-nav-expand">
          <button
            onClick={handleMenuToggleClick}
            className={buttonClasses()}
            aria-controls="side-nav-wrapper"
            aria-expanded={menuIsOpen}
          >
            <Menu size={20} role="presentation" focusable="false" />{" "}
            <span className="has-black-text">Explore</span>
          </button>
        </div>

        <nav data-accordion="side-nav-accordion" className={menuClasses()} id="side-nav-wrapper">
          <p className="version-text has-no-padding has-gray800-text is-xsmall-12 column">
            Version {pkg.version}
          </p>
          {renderNavAccordion()}
        </nav>
      </div>
    </div>
  )
}
