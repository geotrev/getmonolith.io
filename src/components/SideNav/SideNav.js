import React, { Component } from "react"
import "./SideNav.scss"

const path = require("path")
const pkg = require("../../../package.json")

import classNames from "classnames"
import { NavLink } from "react-router-dom"
import Menu from "react-feather/dist/icons/menu"
import ChevronRight from "react-feather/dist/icons/chevron-right"

import { Button } from "components"
import Undernet from "undernet"

export default class SideNav extends Component {
  constructor(props) {
    super(props)
    this.state = { menuIsOpen: null }
    this.getWidth = this.getWidth.bind(this)
    this.handleClick = this.handleClick.bind(this)
    this.handleCollapseClick = this.handleCollapseClick.bind(this)
  }

  componentWillMount() {
    this.getWidth()
    window.addEventListener("resize", this.getWidth)

    if (window.outerWidth < 1200) {
      this.setState({ menuIsOpen: false })
    } else {
      this.setState({ menuIsOpen: true })
    }
  }

  componentDidMount() {
    Undernet.Accordions.start()
  }

  componentDidUpdate() {
    Undernet.Accordions.stop()
    Undernet.Accordions.start()
  }

  componentWillUnmount() {
    window.removeEventListener("resize", this.getWidth)
    Undernet.Accordions.stop()
  }

  handleCollapseClick() {
    if (window.outerWidth > 1199) return
    this.setState({ menuIsOpen: false })
  }

  getWidth() {
    if (window.outerWidth > 1199) {
      this.setState({ menuIsOpen: true })
    }
  }

  handleClick(e) {
    e.preventDefault()
    this.setState({ menuIsOpen: !this.state.menuIsOpen })
  }

  render() {
    const buttonClasses = classNames(
      "is-justified-center is-aligned-center is-flex is-hidden-xlarge",
      {
        "rotate-180": !this.state.menuIsOpen,
      },
    )

    const menuClasses = classNames("row collapsed side-nav-menu accordion", {
      "is-hidden": !this.state.menuIsOpen,
    })

    return (
      <div className="xsmall-12 xlarge-2 columns collapsed" id="side-nav">
        <div className="fluid grid side-nav-wrapper">
          <div className="row is-flex is-hidden-xlarge side-nav-expand">
            <Button onClick={this.handleClick} href="#" className={buttonClasses}>
              <Menu size={20} /> <span className="has-black-text">Explore</span>
            </Button>
          </div>

          <nav data-accordion="nav-accordion" className={menuClasses}>
            <p className="version-text has-no-margin has-gray800-text xsmall-12 columns">
              Version {pkg.version}
            </p>
            {this.renderLists()}
          </nav>
        </div>
      </div>
    )
  }

  accordionIsActive(items) {
    let isActive = false

    items.forEach(item => {
      if (window.location.href.includes(item.url)) {
        isActive = true
      }
    })

    return isActive
  }

  renderLists() {
    return this.props.navItems.map((section, i) => {
      let listItems = section.links.map((item, j) => {
        return (
          <li key={j}>
            <NavLink
              className="has-black-text is-flex is-aligned-center"
              activeClassName="active"
              onClick={this.handleCollapseClick}
              to={item.url}
            >
              {item.name} <ChevronRight size={16} />
            </NavLink>
          </li>
        )
      })

      return (
        <div
          data-expanded={this.accordionIsActive(section.links) ? "true" : "false"}
          data-accordion-row={"nav-acc-content" + i}
          className={classNames("accordion-row", this.props.navListClasses)}
          key={i}
        >
          <ul>
            <li>
              <h4 id={"nav-acc-button" + i} className="paragraph">
                <button
                  data-parent="nav-accordion"
                  className="accordion-button"
                  data-target={"nav-acc-content" + i}
                >
                  {section.header}
                </button>
              </h4>
            </li>
            <li
              className="accordion-content"
              id={"nav-acc-content" + i}
              aria-labelledby={"nav-acc-button" + i}
              data-content={this.accordionIsActive(section.links) ? "visible" : "hidden"}
            >
              <ul>{listItems}</ul>
            </li>
          </ul>
        </div>
      )
    })
  }
}
