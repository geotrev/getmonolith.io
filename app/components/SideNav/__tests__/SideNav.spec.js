import React from "react"
import SideNav from "../SideNav"
import { BrowserRouter as Router } from "react-router-dom"
import { Accordions } from "undernet"

jest.mock("react-feather/dist/icons/chevron-right", () => global.simpleMock("ChevronRight"))
jest.mock("react-feather/dist/icons/menu", () => global.simpleMock("Menu"))
jest.mock("projectRoot/package.json", () => ({
  version: "9.9.9",
}))

Accordions.start = jest.fn()
Accordions.stop = jest.fn()

const MENU_COLLAPSE_WIDTH = 1199
const MENU_EXPAND_WIDTH = MENU_COLLAPSE_WIDTH + 1
const DEFAULT_WIDTH = 1024
const navItems = [
  {
    header: "Test Header",
    links: [{ name: "Test 1", url: "#" }],
  },
]

function mountComponent() {
  return mount(
    <Router>
      <SideNav navItems={navItems} />
    </Router>
  )
}

function updatePageWidth(width) {
  global.window.innerWidth = width
  window.dispatchEvent(new Event("resize"))
}

describe.only("<SideNav />", () => {
  describe("#render", () => {
    it("renders", () => {
      // Given
      const wrapper = mountComponent()
      // Then
      expect(wrapper.find("SideNav")).toMatchSnapshot()
    })
  })

  describe("#state", () => {
    beforeEach(() => {
      const wrapper = mountComponent()
      wrapper.find("SideNav").setState({ menuIsOpen: true })
    })

    it("calls Accordions.stop if state.menuIsOpen changes", () => {
      expect(Accordions.stop).toHaveBeenCalled()
    })

    it("calls Accordions.start if state.menuIsOpen changes", () => {
      expect(Accordions.start).toHaveBeenCalled()
    })
  })

  describe("#handleMenuVisibility", () => {
    afterEach(() => {
      updatePageWidth(DEFAULT_WIDTH)
    })

    it(`hides menu at or below ${MENU_COLLAPSE_WIDTH}`, () => {
      // Given
      const wrapper = mountComponent()
      // Then
      expect(wrapper.find(".side-nav-wrapper")).toMatchSnapshot()
    })

    it(`shows menu above ${MENU_COLLAPSE_WIDTH}`, () => {
      // Given
      updatePageWidth(MENU_EXPAND_WIDTH)
      const wrapper = mountComponent()
      // Then
      expect(wrapper.find(".side-nav-wrapper")).toMatchSnapshot()
    })
  })

  describe("#handleMenuToggleClick", () => {
    it("toggles menu when button is clicked", () => {
      // Given
      const wrapper = mountComponent()
      const button = wrapper.find(".side-nav-expand button")
      // When
      button.simulate("click")
      // Then
      expect(wrapper.find(".side-nav-wrapper")).toMatchSnapshot()
    })
  })
})
