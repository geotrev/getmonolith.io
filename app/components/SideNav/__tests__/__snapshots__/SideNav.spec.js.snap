// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNav /> #handleMenuToggleClick toggles menu when button is clicked 1`] = `
<div
  className="is-fluid grid side-nav-wrapper"
>
  <div
    className="row has-display-flex has-display-none-xlarge side-nav-expand"
  >
    <button
      aria-controls="side-nav-wrapper"
      aria-expanded={true}
      className="has-justify-content-center has-align-items-center has-display-flex has-display-none-xlarge"
      onClick={[Function]}
    >
      <Menu
        focusable="false"
        role="presentation"
        size={20}
      />
       
      <span
        className="has-black-text"
      >
        Explore
      </span>
    </button>
  </div>
  <nav
    className="row side-nav-menu accordion has-padding-3"
    data-accordion="side-nav-accordion"
    id="side-nav-wrapper"
  >
    <p
      className="version-text has-no-padding has-gray800-text is-xsmall-12 column"
    >
      Version 
      9.9.9
    </p>
    <div
      className="accordion-row is-xsmall-12 column has-no-padding"
      data-accordion-row="nav-acc-content0"
      data-visible="false"
      key="#"
    >
      <h4
        className="paragraph"
      >
        <button
          className="accordion-button"
          data-parent="side-nav-accordion"
          data-target="nav-acc-content0"
          id="nav-acc-button0"
        >
          Test Header
        </button>
      </h4>
      <ul
        className="accordion-content"
        id="nav-acc-content0"
      >
        <li
          key="Test 1"
          role="none"
        >
          <NavLink
            activeClassName="active"
            className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
            onClick={[Function]}
            role="listitem"
            to="#"
          >
            <Link
              aria-current={null}
              className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
              onClick={[Function]}
              role="listitem"
              to="#"
            >
              <a
                aria-current={null}
                className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
                href="/"
                onClick={[Function]}
                role="listitem"
              >
                Test 1
                 
                <ChevronRight
                  focusable="false"
                  role="presentation"
                  size={16}
                />
              </a>
            </Link>
          </NavLink>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`<SideNav /> #handleMenuVisibility hides menu at or below 1199 1`] = `
<div
  className="is-fluid grid side-nav-wrapper"
>
  <div
    className="row has-display-flex has-display-none-xlarge side-nav-expand"
  >
    <button
      aria-controls="side-nav-wrapper"
      aria-expanded={false}
      className="has-justify-content-center has-align-items-center has-display-flex has-display-none-xlarge rotate-180"
      onClick={[Function]}
    >
      <Menu
        focusable="false"
        role="presentation"
        size={20}
      />
       
      <span
        className="has-black-text"
      >
        Explore
      </span>
    </button>
  </div>
  <nav
    className="row side-nav-menu accordion has-padding-3 has-display-none"
    data-accordion="side-nav-accordion"
    id="side-nav-wrapper"
  >
    <p
      className="version-text has-no-padding has-gray800-text is-xsmall-12 column"
    >
      Version 
      9.9.9
    </p>
    <div
      className="accordion-row is-xsmall-12 column has-no-padding"
      data-accordion-row="nav-acc-content0"
      data-visible="false"
      key="#"
    >
      <h4
        className="paragraph"
      >
        <button
          className="accordion-button"
          data-parent="side-nav-accordion"
          data-target="nav-acc-content0"
          id="nav-acc-button0"
        >
          Test Header
        </button>
      </h4>
      <ul
        className="accordion-content"
        id="nav-acc-content0"
      >
        <li
          key="Test 1"
          role="none"
        >
          <NavLink
            activeClassName="active"
            className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
            onClick={[Function]}
            role="listitem"
            to="#"
          >
            <Link
              aria-current={null}
              className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
              onClick={[Function]}
              role="listitem"
              to="#"
            >
              <a
                aria-current={null}
                className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
                href="/"
                onClick={[Function]}
                role="listitem"
              >
                Test 1
                 
                <ChevronRight
                  focusable="false"
                  role="presentation"
                  size={16}
                />
              </a>
            </Link>
          </NavLink>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`<SideNav /> #handleMenuVisibility shows menu above 1199 1`] = `
<div
  className="is-fluid grid side-nav-wrapper"
>
  <div
    className="row has-display-flex has-display-none-xlarge side-nav-expand"
  >
    <button
      aria-controls="side-nav-wrapper"
      aria-expanded={true}
      className="has-justify-content-center has-align-items-center has-display-flex has-display-none-xlarge"
      onClick={[Function]}
    >
      <Menu
        focusable="false"
        role="presentation"
        size={20}
      />
       
      <span
        className="has-black-text"
      >
        Explore
      </span>
    </button>
  </div>
  <nav
    className="row side-nav-menu accordion has-padding-3"
    data-accordion="side-nav-accordion"
    id="side-nav-wrapper"
  >
    <p
      className="version-text has-no-padding has-gray800-text is-xsmall-12 column"
    >
      Version 
      9.9.9
    </p>
    <div
      className="accordion-row is-xsmall-12 column has-no-padding"
      data-accordion-row="nav-acc-content0"
      data-visible="false"
      key="#"
    >
      <h4
        className="paragraph"
      >
        <button
          className="accordion-button"
          data-parent="side-nav-accordion"
          data-target="nav-acc-content0"
          id="nav-acc-button0"
        >
          Test Header
        </button>
      </h4>
      <ul
        className="accordion-content"
        id="nav-acc-content0"
      >
        <li
          key="Test 1"
          role="none"
        >
          <NavLink
            activeClassName="active"
            className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
            onClick={[Function]}
            role="listitem"
            to="#"
          >
            <Link
              aria-current={null}
              className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
              onClick={[Function]}
              role="listitem"
              to="#"
            >
              <a
                aria-current={null}
                className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
                href="/"
                onClick={[Function]}
                role="listitem"
              >
                Test 1
                 
                <ChevronRight
                  focusable="false"
                  role="presentation"
                  size={16}
                />
              </a>
            </Link>
          </NavLink>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`<SideNav /> #render renders 1`] = `
<SideNav>
  <div
    className="is-xsmall-12 is-xlarge-2 column has-no-padding"
    id="side-nav"
  >
    <div
      className="is-fluid grid side-nav-wrapper"
    >
      <div
        className="row has-display-flex has-display-none-xlarge side-nav-expand"
      >
        <button
          aria-controls="side-nav-wrapper"
          aria-expanded={false}
          className="has-justify-content-center has-align-items-center has-display-flex has-display-none-xlarge rotate-180"
          onClick={[Function]}
        >
          <Menu
            focusable="false"
            role="presentation"
            size={20}
          />
           
          <span
            className="has-black-text"
          >
            Explore
          </span>
        </button>
      </div>
      <nav
        className="row side-nav-menu accordion has-padding-3 has-display-none"
        data-accordion="side-nav-accordion"
        id="side-nav-wrapper"
      >
        <p
          className="version-text has-no-padding has-gray800-text is-xsmall-12 column"
        >
          Version 
          9.9.9
        </p>
        <div
          className="accordion-row is-xsmall-12 column has-no-padding"
          data-accordion-row="nav-acc-content0"
          data-visible="false"
          key="#"
        >
          <h4
            className="paragraph"
          >
            <button
              className="accordion-button"
              data-parent="side-nav-accordion"
              data-target="nav-acc-content0"
              id="nav-acc-button0"
            >
              Test Header
            </button>
          </h4>
          <ul
            className="accordion-content"
            id="nav-acc-content0"
          >
            <li
              key="Test 1"
              role="none"
            >
              <NavLink
                activeClassName="active"
                className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
                onClick={[Function]}
                role="listitem"
                to="#"
              >
                <Link
                  aria-current={null}
                  className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
                  onClick={[Function]}
                  role="listitem"
                  to="#"
                >
                  <a
                    aria-current={null}
                    className="side-nav-link-item has-black-text is-display-flex has-align-items-center"
                    href="/"
                    onClick={[Function]}
                    role="listitem"
                  >
                    Test 1
                     
                    <ChevronRight
                      focusable="false"
                      role="presentation"
                      size={16}
                    />
                  </a>
                </Link>
              </NavLink>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </div>
</SideNav>
`;
